gnocchi/cli/metricd.py

```diff
@@ -178,7 +221,13 @@ def _run_job(self):
                           exc_info=True)
             finally:
                 lock.release()
+            # Discard the sack from the sacks since we just did it.
+            self.sacks_with_measures_to_process.discard(s)
```

Comment:
should this be cleared before we process it? or else it could remove a newly added entry between us processing and getting to this line.

Reply:
Yes, you're right, but, if we more it before, and an exception occurs in the processing, then, it'll be removed and it won't be processed again. So here I've made the trade-off of saying, well, you can be processing measures files 1, 2, 3, and while you're doing it, there's 4 coming, but you missed it, well, it'll be processed next interval _anyway_.

Reply:
can we add it back if there's an exception? or we worried that it's bad to blindly add something back?

Reply:
I think we can add it back if we make sure it was there in the first place, of course. Modified the patch that way.